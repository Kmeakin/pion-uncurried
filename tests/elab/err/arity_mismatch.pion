let f = fn(x: Bool) => x;

let x = f();

let y = f(true, false);