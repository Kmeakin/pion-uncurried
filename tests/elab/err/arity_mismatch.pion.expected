let f: fn(x: Bool) -> Bool = fn(x: Bool) => x;

let x: #error = #error;

let y: #error = #error;

====DIAGNOSTICS====
Error: Called function with too many arguments
   ╭─[tests/elab/err/arity_mismatch.pion:5:9]
   │
 5 │ let y = f(true, false);
   ·         │ 
   ·         ╰─ Help: this function expects 1 argument but you gave it 2
   ·         │ 
   ·         ╰─ Help: type of this function is `fn(x: Bool) -> Bool`
───╯
Error: Called function with too few arguments
   ╭─[tests/elab/err/arity_mismatch.pion:3:9]
   │
 3 │ let x = f();
   ·         │ 
   ·         ╰─ Help: this function expects 1 argument but you gave it 0
   ·         │ 
   ·         ╰─ Help: type of this function is `fn(x: Bool) -> Bool`
───╯
